README.md

```diff
@@ -96,8 +96,9 @@ Here are the available options, in some vague order of relevance to most common
 - An `Array` of `Exception` classes (retry any exception of one of these types, including subclasses)
 - A `Hash` where the keys are `Exception` classes and the values are one of:
   - `nil` (retry every exception of the key's type, including subclasses)
+  - A single `Proc` (retries exceptions ONLY if it returns truthy)
   - A single `Regexp` pattern (retries exceptions ONLY if their `message` matches the pattern)
-  - An array of patterns (retries exceptions ONLY if their `message` matches at least one of the patterns)
+  - An array of `Proc` and/or `Regexp` (retries exceptions ONLY if at least one exception matches `Regexp` or the `Proc` evaluates to `true`)
```

Comment:
grammar: "retries exceptions ONLY if at least one exception message matches a `Regexp` or at least one `Proc` evaluates to `true`" (or "returns `truthy`"... not sure which it is but you should be correct and consistent in the documentation).